### PR TITLE
Add BooleanArray type

### PR DESCRIPTION
**Pull request by ajgarlag**

### CODE REVIEW

**Comment by ajgarlag** [comment]
@martin-georgiev friendly ping here. Could you review this PR?

**Comment by martin-georgiev** [comment]
Sorry that it took a lot of time to look at this @ajgarlag. It seems the QA tools can't validate the change. Do you mind having another look?

File: src/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArray.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+
+/**
+ * Implementation of PostgreSql BOOL[] data type.
+ *
+ * @see https://www.postgresql.org/docs/9.4/static/arrays.html
+ * @since 1.5.3
+ *
+ * @author Martin Georgiev <martin.georgiev@gmail.com>
```
Comment (martin-georgiev) [inline src/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArray.php:15]: You can include your authorship details here.

Comment (ajgarlag) [reply src/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArray.php:15]: Done, thanks!

**Comment by ajgarlag** [comment]
@martin-georgiev I've rebased and fixed the code. I hope it can be merged soon. Waiting for your review.

File: tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    /**
+     * @var BooleanArray|MockObject
+     */
+    private $fixture;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->platform = new PostgreSQL94Platform();
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php:28]:
```suggestion
        $this->platform = $this->createMock(AbstractPlatform::class);
```

File: tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
+use MartinGeorgiev\Doctrine\DBAL\Types\BooleanArray;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class BooleanArrayTest extends TestCase
+{
+    /**
+     * @var PostgreSQL94Platform
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php:15]:
```suggestion
     * @var AbstractPlatform|MockObject
```

File: tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\MartinGeorgiev\Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php:7]:
```suggestion
use Doctrine\DBAL\Platforms\AbstractPlatform;
```

File: src/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArray.php (hunk 1 above)
Comment (martin-georgiev) [reply src/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArray.php:15]:
```suggestion
```

File: tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php (hunk 2 above)
Comment (ajgarlag) [reply tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php:28]: The new `BooleanArray` type leverages `PostgreSQL94Platform::convertBooleansToDatabaseValue` and `PostgreSQL94Platform::convertFromBoolean`.

If we make this change, we have to mock these two methods increasing tests complexity and adding a third parameter to `BooleanArrayTest::validTransformations` data provider to provide the platform value.

Please, confirm that it is what you want. Thanks.

Comment (ajgarlag) [reply tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php]: I've implemented this in a separate commit, so you can evaluate it.

Comment (martin-georgiev) [reply tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php]: > The new `BooleanArray` type leverages `PostgreSQL94Platform::convertBooleansToDatabaseValue` and `PostgreSQL94Platform::convertFromBoolean`.
> 
> If we make this change, we have to mock these two methods increasing tests complexity and adding a third parameter to `BooleanArrayTest::validTransformations` data provider to provide the platform value.
> 
> Please, confirm that it is what you want. Thanks.

Ah, I did not spot this earlier. Thanks for explaining your reasoning - it makes sense to me now.

**Review by martin-georgiev — APPROVED** [review]

**Comment by martin-georgiev** [comment]
Cheers for contributing to the project. This is now included in https://github.com/martin-georgiev/postgresql-for-doctrine/releases/tag/v1.6.0

**Pull request merged**